pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -0,0 +1,239 @@
+	if err != nil {
+		return err
+	}
+
+	// Collect the current latency profile
+	configNodeObj, err := c.nodeLister.Get("cluster")
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	} else if errors.IsNotFound(err) {
+		// if config/v1/node/cluster object is not found this controller should do nothing
+		return nil
+	}
+
+	condition := metav1.Condition{
+		Type:   apiconfigv1.KubeControllerManagerDegraded,
+		Status: metav1.ConditionUnknown,
+	}
+	// do nothing if workerLatencyProfile is not set
+	if configNodeObj.Spec.WorkerLatencyProfile == "" {
+		// TODO: in case worker latency profile is transitioned
+		// from Default/Medium/Low to ""
+		// check that the required extendedArgs should vanish
+
+		condition.Reason = reasonWorkerLatencyProfileEmpty
+		condition.Message = "worker latency profile not set on cluster"
+		condition.Status = metav1.ConditionFalse
+		_, err := c.updateConfigNodeStatus(condition)
+		return err
+	}
+
+	desiredControllerManagerArgumentVals := map[string]string{}
+	switch configNodeObj.Spec.WorkerLatencyProfile {
+	case apiconfigv1.DefaultUpdateDefaultReaction:
+		desiredControllerManagerArgumentVals[nodeMonitorGracePeriodArgument] = strconv.Itoa(int(apiconfigv1.DefaultNodeMonitorGracePeriod))
+	case apiconfigv1.MediumUpdateAverageReaction:
+		desiredControllerManagerArgumentVals[nodeMonitorGracePeriodArgument] = strconv.Itoa(int(apiconfigv1.MediumNodeMonitorGracePeriod))
+	case apiconfigv1.LowUpdateSlowReaction:
+		desiredControllerManagerArgumentVals[nodeMonitorGracePeriodArgument] = strconv.Itoa(int(apiconfigv1.LowNodeMonitorGracePeriod))
+	}
+
+	// Collect the unique set of revisions of the node static pods
+	revisionMap := map[int32]struct{}{}
+	uniqueRevisions := []int32{}
+	for _, nodeStatus := range operatorStatus.NodeStatuses {
+		revision := nodeStatus.CurrentRevision
+		if _, ok := revisionMap[revision]; !ok {
+			revisionMap[revision] = struct{}{}
+			uniqueRevisions = append(uniqueRevisions, revision)
+		}
+	}
+
+	// For each revision, check that the configmap for that revision have
+	// correct argument values or not
+	revisionsHaveSynced := true
+	for _, revision := range uniqueRevisions {
+		configMapNameWithRevision := fmt.Sprintf("%s-%d", controllerManagerConfigMapName, revision)
+		configMap, err := c.configMapClient.ConfigMaps(operatorclient.TargetNamespace).Get(ctx, configMapNameWithRevision, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		matches := true
+		for arg, val := range desiredControllerManagerArgumentVals {
+			match, err := configMatchesControllerManagerArgument(configMap, arg, val)
+			if err != nil {
+				return err
+			}
+			matches = matches && match
+		}
+		if !matches {
+			revisionsHaveSynced = false
+			break
+		}
+	}
+
+	conditions := []metav1.Condition{
+		{
+			Type:   apiconfigv1.KubeControllerManagerComplete,
+			Status: metav1.ConditionUnknown,
+			Reason: reasonWorkerLatencyProfileUpdated,
```

Comment:
the reasons should at least flip once when transitioning between `False -> True` and `True -> False`

Reply:
Added three **reasons** for that reason :)
1. ProfileEmpty
2. ProfileUpdateTriggered
3. ProfileUpdated

Cases
- 1: When worker latency profile is not set on cluster i.e when config node object has an empty spec
- 2: When worker latency profile rollout is triggered by the config observers i.e. right after this status should be KCMProgressing
- 3: When worker latency profile rollout is complete, as all current revisions have the required latency profile arg(s) applied i.e. right when we're setting KCMComplete

@atiratree Let me know if this needs any further change including any changes in the updated implementation for status updates.